utils/cloud_constants.py

```diff
@@ -83,3 +86,5 @@
 OSA_API_SERVER_PORT = os.environ.get("OSA_API_SERVER_PORT", 5000)
 OSA_API_SERVER_URL = 'http://{host}:{port}'.format(host=OSA_API_SERVER_HOST, port=OSA_API_SERVER_PORT)
 DATA_INSERT_CONCURRENCY = int(os.environ.get("DATA_INSERT_CONCURRENCY", 10))
+# Shared instance of S3FileSystem
+INFERENCE_S3FS = S3FileSystem(key=AWS_INFERENCE_S3_ACCESS_KEY_ID, secret=AWS_INFERENCE_S3_SECRET_ACCESS_KEY)
```

Comment:
Pandas uses s3fs internally when you provide a s3:// url, is there any reason we need to use it explicitly?

Reply:
Oh nevermind I just understood what the issue is.

Reply:
@rootAvish 

By default s3fs infers the AWS credentials through the env variables {AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY}. In our case we have two AWS S3 resources with two different credentials to access it. So we can't globally set {AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY} and expect it to function.

So instead of letting the pandas to create instance of S3FileSystem with default credentials, explicit instance being used to read/write into S3 with appropriate credentials.